src/backend/joanie/tests/core/test_api_course_product_relations.py

```diff
@@ -1504,3 +1505,64 @@ def test_api_course_product_relation_payment_schedule_with_product_id_anonymous(
 
         self.assertEqual(response_relation_path.status_code, HTTPStatus.OK)
         self.assertEqual(response_relation_path.json(), response.json())
+
+    @override_settings(
+        JOANIE_PAYMENT_SCHEDULE_LIMITS={
+            5: (100,),
+        },
+        DEFAULT_CURRENCY="EUR",
+    )
+    def test_api_course_product_relation_payment_schedule_with_certificate_product_id_anonymous(
+        self,
+    ):
+        """
+        Anonymous users should be able to retrieve a payment schedule for
+        a single course product relation if a product id is provided
+        and the product is a certificate.
+        """
+        course_run = factories.CourseRunFactory(
+            enrollment_start=datetime(2024, 1, 1, 14, tzinfo=ZoneInfo("UTC")),
+            start=datetime(2024, 3, 1, 14, tzinfo=ZoneInfo("UTC")),
+            end=datetime(2024, 5, 1, 14, tzinfo=ZoneInfo("UTC")),
+        )
+        product = factories.ProductFactory(
+            price=3,
+            type=enums.PRODUCT_TYPE_CERTIFICATE,
+        )
+        relation = factories.CourseProductRelationFactory(
+            course=course_run.course,
+            product=product,
+            organizations=factories.OrganizationFactory.create_batch(2),
+        )
+
+        with (
+            mock.patch("uuid.uuid4", return_value=uuid.UUID(int=1)),
```

Comment:
What the point to patch it ? Is it needed ? I don't see later where it is used. And you probably patch it with a real uuid v4 ?